haystack/nodes/connector/crawler.py

```diff
@@ -151,11 +151,15 @@ def crawl(
             logger.info(f"Fetching from {urls} to `{output_dir}`")
             sub_links: Dict[str, List] = {}
 
-            # don't go beyond the initial list of urls
-            if crawler_depth == 0:
```

Comment:
Yes. In this PR I refactored the crawler tests (they were by far the slowest suite and easy to mock), and I caught a small bug in the process, so I fixed it. Should have done it separately, but at the beginning I though this PR would be mostly about fixing the tests, rather than the CI itself :smile: